test/components/app_test.js

```diff
@@ -2,30 +2,56 @@ import {renderComponent, expect} from '../test_helper';
 import App from '../../src/app/components/app';
 import FireBaseTools from '../../src/app/utils/firebase';
 
-describe('App', () => {
-  let component;
-
-  beforeEach(() => {
-    component = renderComponent(App);
-  });
-
-  it('renders something', () => {
-    expect(component).to.exist;
-  });
-});
-
 describe('fillNoClassThisDay', () => {
   let noClassThisDay = new Array();
-
   FireBaseTools.fillNoClassThisDay(noClassThisDay);
+
+  it('expect array', () => {
+    expect(noClassThisDay).to.be.a('array');
+  });
   it('fill', () => {
       expect(noClassThisDay.length).greaterThan(0);
   });
   it('number of days', () => {
     expect(noClassThisDay.length).equals(11);
   });
-    it('number of days', () => {
-    expect(noClassThisDay.length).equals(11);
+});
+
+describe('Weekdays', () => {
+  let startDate = new Date(2017, 1, 9);
+  let endDate = new Date(2017, 4, 13);
+  let noClassThisDay = new Array();
+  FireBaseTools.fillNoClassThisDay(noClassThisDay);
+  let val = () => { return {
```

Comment:
We need to get snap.val() for the function checkWeekday and so in order to do this we make val a lambda function and make snap is a variable containing a lambda function.